packages/contracts-sdk/src/implementations/farcaster/FarcasterIdGatewayContract.ts

```diff
@@ -0,0 +1,199 @@
+import { ID_GATEWAY_EIP_712_TYPES } from "@farcaster/hub-nodejs";
+import {
+  EVMAccountAddress,
+  EVMContractAddress,
+  BlockchainCommonErrors,
+  FarcasterIdGatewayContractError,
+  UnixTimestamp,
+  SignedRegisterSignature,
+} from "@snickerdoodlelabs/objects";
+import { TypedDataField, ethers } from "ethers";
+import { injectable } from "inversify";
+import { ResultAsync, okAsync } from "neverthrow";
+
+import { BaseContract } from "@contracts-sdk/implementations/BaseContract.js";
+import { IEthersContractError } from "@contracts-sdk/implementations/BlockchainErrorMapper.js";
+import {
+  ContractOverrides,
+  IFarcasterIdGatewayContract,
+  WrappedTransactionResponse,
+} from "@contracts-sdk/interfaces/index.js";
+import { ContractsAbis } from "@contracts-sdk/interfaces/objects/index.js";
+
+@injectable()
+export class FarcasterIdGatewayContract
+  extends BaseContract<FarcasterIdGatewayContractError>
+  implements IFarcasterIdGatewayContract
+{
+  constructor(
+    protected providerOrSigner: ethers.Provider | ethers.Signer,
+    protected contractAddress: EVMContractAddress,
+  ) {
+    super(
+      providerOrSigner,
+      contractAddress,
+      ContractsAbis.FarcasterIdGatewayAbi.abi,
+    );
+  }
+
+  public price(
+    extraStorage?: bigint,
+  ): ResultAsync<
+    bigint,
+    FarcasterIdGatewayContractError | BlockchainCommonErrors
+  > {
+    // If extraStorage was provided, call price with it, otherwise call without
+    // https://optimistic.etherscan.io/address/0x00000000fc25870c6ed6b6c7e41fb078b7656f69#code#F2#L64
+    return ResultAsync.fromPromise(
+      extraStorage
```

Comment:
Can you add the specific != undefined on this? Bare null checks have a way of biting you in the ass.